transifex/openformats
# Add option to remove empty strings in YAML

**TODO**
- [x] Remove comments of deleted entries
- [x] Optimize template rebuilding to reduce memory footprint

Problem and/or solution
-----------------------
We need to be able to remove entries with no translation but at the same time use the template.

The previous implementation, when instructed to only use the entries that had a translation, didn't use the template at all. Instead, it generated a new document from scratch, thus losing the initial structure, comments etc.

With the approach this PR is introducing, there is no need to ditch the template anymore in such cases, so the resulting document has the same structure as before.

How to test
-----------
Unit tests.

Manual testing through Transifex (related PRs pending)
- upload YAML resource
- translate some strings only
- download for use
- make sure that:
  - is valid YAML
  - untranslated strings are not present in the file
  - the overall structure is correct
    - only empty strings and their corresponding comments are missing
    - parent dictionaries of removed strings are not deleted
  - anything else that I may have missed!

Reviewer checklist
------------------

Code:

PR:
